streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -365,17 +365,47 @@ public void handleAssignment(final Map<TaskId, Set<TopicPartition>> activeTasks,
         // 1. for tasks that are already owned, just update input partitions / resume and skip re-creating them
         // 2. for tasks that have changed active/standby status, just recycle and skip re-creating them
         // 3. otherwise, close them since they are no longer owned
+        final Map<TaskId, RuntimeException> failedTasks = new LinkedHashMap<>();
         if (stateUpdater == null) {
             handleTasksWithoutStateUpdater(activeTasksToCreate, standbyTasksToCreate, tasksToRecycle, tasksToCloseClean);
         } else {
-            handleTasksWithStateUpdater(activeTasksToCreate, standbyTasksToCreate, tasksToRecycle, tasksToCloseClean);
+            final Map<Task, Set<TopicPartition>> tasksToRecycleFromStateUpdater = new HashMap<>();
```

Comment:
I'd put this block inside `handleTasksWithStateUpdater` to separate it from the non-state updater code